bin/syclcc-clang

```diff
@@ -448,135 +494,239 @@ def run_or_print(command, print_only):
-    self._compiler_args += config.common_compiler_args
+        linker_args += ["-Xclang"]
+    linker_args += ["-fopenmp"]
+
+    return linker_args
+
+# This is a workaround to have access to a backend
+# that can execute host tasks when compiling for GPUs.
+# It should be removed once we have non-OpenMP host backends
+# (e.g. TBB)
+class omp_sequential_invocation:
+  def __init__(self, config):
+    self._cxx = config.pure_cpu_compiler
+
+  def get_compiler_preference(self):
+    return (self._cxx, 1)
 
-    self._linker_args += [
-      "-L"+hipsycl_library_path,
-      "-lhipSYCL-rt",
+  def get_cxx_flags(self):
+
+    flags = [
+        "-D__HIPSYCL_ENABLE_OMPHOST_TARGET__",
+      ]
+      
+    return flags
+
+  def get_linker_flags(self):
+    linker_args = [
       "-lboost_context",
-      "-lboost_fiber"
+      "-lboost_fiber",
+      "-lomp"
     ]
-    if "clang" in os.path.basename(self._compiler):
-      self._linker_args += ["-rpath",hipsycl_library_path]
+    
+    return linker_args
 
-    self._compiler_args += [
-      "-I" + os.path.join(config.hipsycl_installation_path, "include/"),
-      "-I" + os.path.join(config.hipsycl_installation_path, "include/hipSYCL/contrib")
+class compiler:
+  def __init__(self, config):
+    self._user_args = config.forwarded_compiler_arguments
+    self._requires_linking = config.contains_linking_stage()
+    self._requires_compilation = not config.is_pure_linking_stage()
+    self._is_dry_run = config.is_dryrun
+    self._targets = config.targets
+    self._common_compiler_args = config.common_compiler_args
+    self._hipsycl_path = config.hipsycl_installation_path
+    self._hipsycl_lib_path = os.path.join(self._hipsycl_path, "lib/")
+
+    if "hip" in self._targets and "cuda" in self._targets:
+      raise RuntimeError("CUDA and HIP cannot be targeted simultaneously")
+    # clang bug prevents clang -x cuda -fopenmp from working
+    if "omp" in self._targets and "cuda" in self._targets:
+      raise RuntimeError("CUDA and OpenMP cannot be targeted simultaneously")
+
+    self._backends = []
+
+    for backend in self._targets:
+      if backend == "omp":
+        self._backends.append(omp_invocation(config))
+      elif backend == "cuda":
+        self._backends.append(cuda_invocation(config))
+      elif backend == "hip":
+        self._backends.append(hip_invocation(config))
+      else:
+        raise RuntimeError("Unknown backend: " + backend)
+      
+    
+    if not "omp" in self._targets:
+      # We need at least OpenMP "lite" (i.e. without -fopenmp) to
+      # get access to things like host tasks
+     self._backends.append(omp_sequential_invocation(config))
+
+  @property
+  def common_cxx_flags(self):
+    args = [
+      "-I" + os.path.join(self._hipsycl_path, "include/"),
+      "-D__HIPSYCL__"
+    ] + self._common_compiler_args
+
+    return args
+
+  @property
+  def common_linker_flags(self):
+    linker_args = [
+      "-Wl,-rpath", self._hipsycl_lib_path,
```

Comment:
I think this line should be `"-Wl,-rpath=" + self._hipsycl_lib_path,` because currently syclcc emits arguments in the following form: ` -Wl,-rpath /opt/hipSYCL/bin/../lib/` It seems that clang (10.0.1) does not forward the path, but only the `-rpath` to the linker causing an error, since clang tries to compile `/opt/hipSYCL/bin/../lib/`

Reply:
:+1:
and for some reason, I also had to add quotes around the parameter...

Reply:
I can reproduce the issue on my Ubuntu machine here, so I switched to `-rpath=`, but I'm pretty sure that it worked on my Arch Linux machine (but the issue might not trigger depending on argument order, so maybe I just didn't notice).

@fodinabor I don't understand why we would need to add quotes. Is this something that is maybe specific to some linkers?

Reply:
oh sorry.. didn't update.. should be fine this way.. just had to add quotes on the terminal to indicate where the parameter ends, that is not necessary when invoking a process using a parameter array, so should be fine!

Reply:
I see, thanks!

Reply:
linking Issue #437 here